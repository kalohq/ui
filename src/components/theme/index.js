/* @flow */
export {default} from './theme';
