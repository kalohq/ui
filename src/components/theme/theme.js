/* @flow */

import tokens from '../../design-tokens/tokens.theme.js';

/**
 * ThemeProvider Kalo Theme
 * 
 * This is the core CSS-in-JS theme that is passed in to the
 * global ThemeProvider. It is mainly composed of our core
 * design tokens.
 */

const theme = {
  space: [0, 8, 16, 32, 64],
  breakpoints: [],
  ...tokens,
};

export default theme;
