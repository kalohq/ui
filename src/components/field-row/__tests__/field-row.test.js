/* eslint-env jest */
import React from 'react';
import {shallow} from 'enzyme';
import renderer from 'react-test-renderer';
import {ThemeProvider} from 'emotion-theming';

import theme from '../../theme';
import Input from '../../input';
import FieldRow, {DEFAULT_SPACING} from '../field-row.react';
import Field from '../../field';

describe('Input', () => {
  const create = props =>
    shallow(
      <FieldRow {...props}>
        <Field label="Project name">
          <Input placeholder="My first project" />
        </Field>
        <Field label="Task name">
          <Input placeholder="My first task" />
        </Field>
      </FieldRow>
    );

  test('should render shallow component ok', () => {
    const element = renderer
      .create(
        <ThemeProvider theme={theme}>
          <FieldRow>
            <Field label="Project name">
              <Input placeholder="My first project" />
            </Field>
            <Field label="Task name">
              <Input placeholder="My first task" />
            </Field>
          </FieldRow>
        </ThemeProvider>
      )
      .toJSON();
    expect(element).toMatchSnapshot();
  });

  test('should add default spacing to child components', () => {
    const element = create({
      legend: 'Your name',
    });
    expect(
      element
        .find(Field)
        .at(1)
        .props().paddingLeft
    ).toBe(DEFAULT_SPACING);
  });
});
