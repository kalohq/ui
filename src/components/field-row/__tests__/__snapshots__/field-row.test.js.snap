// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input should render shallow component ok 1`] = `
<div
  className="ui-field-row e1q9f2xp6 css-150a2en"
  style={Object {}}
>
  <div
    className="ui-field ui-field--legacy e1q9f2xp6 css-6smw74"
    onClick={undefined}
    style={Object {}}
  >
    <div
      className="e1q9f2xp0 css-1dvaom5"
      style={Object {}}
    >
      <label
        className="ui-field-label e1q9f2xp6 css-4or71o"
        htmlFor={undefined}
        style={Object {}}
        title={undefined}
      >
        Project name
      </label>
    </div>
    <div
      className="ui-input-container ui-input-container--margin-none"
    >
      <input
        className="ui-input ui-input--medium"
        defaultValue={undefined}
        disabled={false}
        name="My first project"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        placeholder="My first project"
        readOnly={false}
        style={
          Object {
            "width": undefined,
          }
        }
        value={undefined}
      />
    </div>
  </div>
  <div
    className="ui-field ui-field--legacy e1q9f2xp6 css-6smw74"
    onClick={undefined}
    style={Object {}}
  >
    <div
      className="e1q9f2xp0 css-1dvaom5"
      style={Object {}}
    >
      <label
        className="ui-field-label e1q9f2xp6 css-4or71o"
        htmlFor={undefined}
        style={Object {}}
        title={undefined}
      >
        Task name
      </label>
    </div>
    <div
      className="ui-input-container ui-input-container--margin-none"
    >
      <input
        className="ui-input ui-input--medium"
        defaultValue={undefined}
        disabled={false}
        name="My first task"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        placeholder="My first task"
        readOnly={false}
        style={
          Object {
            "width": undefined,
          }
        }
        value={undefined}
      />
    </div>
  </div>
</div>
`;
